hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeManager.java

```diff
@@ -71,64 +86,67 @@ public FeeManager(@NonNull final ExchangeRateManager exchangeRateManager) {
      *
      * @param bytes The new fee schedule file content.
      */
-    public void update(@NonNull final Bytes bytes) {
+    public ResponseCodeEnum update(@NonNull final Bytes bytes) {
+        // Parse the current and next fee schedules
+        final CurrentAndNextFeeSchedule schedules;
         try {
-            // Parse the current and next fee schedules
-            final var schedules = CurrentAndNextFeeSchedule.PROTOBUF.parse(bytes.toReadableSequentialData());
-
-            // Get the current schedule
-            var currentSchedule = schedules.currentFeeSchedule();
-            if (currentSchedule == null) {
-                // If there is no current schedule, then we default to the default schedule. Since the default
-                // schedule is completely empty, this will effectively disable the handling of any transactions,
-                // since we don't know what to charge for them.
-                logger.warn("Unable to parse current fee schedule, will default to an empty schedule, effectively"
-                        + "disabling all transactions.");
-                currentSchedule = FeeSchedule.DEFAULT;
-            }
+            schedules = CurrentAndNextFeeSchedule.PROTOBUF.parse(bytes.toReadableSequentialData());
+        } catch (final BufferUnderflowException | IOException ex) {
+            return ResponseCodeEnum.FEE_SCHEDULE_FILE_PART_UPLOADED;
+        }
 
-            // Populate the map of HederaFunctionality -> FeeData for the current schedule
-            this.currentFeeDataMap = new HashMap<>();
-            if (currentSchedule.hasTransactionFeeSchedule()) {
-                populateFeeDataMap(currentFeeDataMap, currentSchedule.transactionFeeScheduleOrThrow());
-            } else {
-                logger.warn("The current fee schedule is missing transaction information, effectively disabling all"
-                        + "transactions.");
-            }
+        // Get the current schedule
+        var currentSchedule = schedules.currentFeeSchedule();
+        if (currentSchedule == null) {
+            // If there is no current schedule, then we default to the default schedule. Since the default
+            // schedule is completely empty, this will effectively disable the handling of any transactions,
+            // since we don't know what to charge for them.
+            logger.warn("Unable to parse current fee schedule, will default to an empty schedule, effectively"
+                    + "disabling all transactions.");
```

Comment:
"disabling all _paid_ transactions."
I assume unpaid (i.e. privileged) transactions are still enabled, so the situation can be remedied (e.g. by updating the fee schedule).

Reply:
I think you are right. The code existed prior to this PR and I did not review it thoroughly.